lib/tests/tests.c

```diff
@@ -638,156 +638,195 @@ test_demographic_events(void)
     ret = tsk_table_collection_init(&tables, 0);
     CU_ASSERT_EQUAL_FATAL(ret, 0);
 
-    for (j = 0; j < n; j++) {
-        samples[j].time = j;
-        samples[j].population_id = j % 2;
-    }
-    ret = msp_alloc(&msp, n, samples, &recomb_map, &tables, rng);
-    CU_ASSERT_EQUAL(ret, 0);
-    /* Zero or negative population sizes are not allowed */
-    ret = msp_set_population_configuration(&msp, 0, -1, 0);
-    CU_ASSERT_EQUAL(ret, MSP_ERR_BAD_PARAM_VALUE);
-    ret = msp_set_population_configuration(&msp, 0, 0, 0);
-    CU_ASSERT_EQUAL(ret, MSP_ERR_BAD_PARAM_VALUE);
+    for (model = 0; model < 2; model++) {
+        for (j = 0; j < n; j++) {
+            samples[j].time = j;
+            /* Ensures DTWF sample times are after demographic events */
+            if (model == 1) {
```

Comment:
I think it's a bug if we need to do this --- there's nothing stopping these happening at the same time, right? This must be a slight issue with the logic in the main loop. Are these events mutually exclusive or something?